frontend/apps/reader/modules/readerzooming.lua

```diff
@@ -371,11 +371,12 @@ function ReaderZooming:onDefineZoom(btn, when_applied_callback)
     if zoom_mode == "columns" or zoom_mode == "rows" then
         if btn ~= "columns" and btn ~= "rows" then
             self.ui:handleEvent(Event:new("SetZoomPan", settings, true))
-            settings.kopt_zoom_factor = self:setNumberOf(
+            config.zoom_factor = self:setNumberOf(
```

Comment:
setNumberOf() does not return anything. This is a bug from @jperon's initial implementation.

This is the cause of the crash in https://github.com/koreader/koreader/pull/7968#issuecomment-878676503

Reply:
Nice catch, thanks ;).

Reply:
With this fix, there is no more crash. However, I actually didn't get what actually changed with all these fixes so far.
Gesture manager's `Zoom to manual` or `Zoom to columns` still zoom to wrong corner and use wrong direction, as before. I still need to tap some directions twice for them to jump to their starting locations.

Reply:
Zoom factor value when changing columns is updating properly now! Thanks :)

Reply:
A side question though: what is the point of showing zoom factor value up to 2 decimal points, when the user can set it up to one decimal point? E.g. it shows actual value in two-columns as 1.88, but in the `...` of `%` it shows 1.9. When I actually Apply this 1.9, the zoom factor value is anyway 1.88. For simplicity, shouldn't we just make the info box value one-decimal too?

Reply:
Another curious non-critical observation. Comparing these two situations:
A) Set zoom to content (zoom factor is 1.00 exactly), go to `%`, manually input 1.9 and Apply (zoom factor becomes 1.9 exactly). If I change to 1.7, it becomes exactly 1.7.
B) Set zoom to content, then zoom to columns, select 2 (zoom factor is 1.88). `%` shows it as 1.9. Now, if I change it to 1.7, it will become 1.68.

The actual zoom corresponds to the values properly in both cases. So I guess, in the `...` menu, there is a fixed step of 0.1 instead of overwriting the whole value?

Reply:
I am sorry for bombarding `Zoom` so actively :D It is not that critical for me, I just want to bring my feedback on the clarity of the UI in order to make it more understandable for the user, if possible. Also it is very interesting to test all these cases.

Reply:
> Zoom factor value when changing columns is updating properly now! Thanks :)

Yep, that's the only thing this was supposed to address.